ropetest/contrib/autoimporttest.py

```diff
@@ -142,6 +142,22 @@ def test_generate_full_cache(self):
         for table in self.importer._dump_all():
             self.assertTrue(len(table) > 0)
 
+    def test_generate_full_cache(self):
+        """The single thread test takes much longer than the multithread test but is easier to debug"""
+        single_thread = False
+        self.importer.generate_modules_cache(single_thread=single_thread)
+        """
+        Create a temporary directory and set permissions to 000
+        """
+        import tempfile, sys
+        with tempfile.TemporaryDirectory() as dir:
+            import os
+            os.chmod(dir, 0)
+            sys.path.append(dir)
```

Comment:
This test lacks isolation, changing the sys.path here is global mutation that won't be undone by the end of the test so it will affect other tests as well.

Reply:
@MrBago I think it's enough to remove the dir at the end of the context manager.

Reply:
@lieryan I updated the test to use `prefs.python_paths` instead of sys.path.